docs/csharp/misc/cs0236.md

```diff
@@ -13,9 +13,27 @@ A field initializer cannot reference the non-static field, method, or property '
 
 Instance fields cannot be used to initialize other instance fields outside a method.
 
+## Why this error occurs
+
+The compiler enforces this restriction because of how object initialization works in C#. When an object is created, field initializers are processed before any constructor code runs. During this phase:
+
+1. **Initialization order is not guaranteed**: The C# specification doesn't define the order in which field initializers are executed. If one field could reference another, the referenced field might not be initialized yet.
```

Comment:
This is false. Field initializers are processed in lexical order in a single file. The only non-guarantee is for `partial` class and struct types, where the order of the fields initializers in different source files isn't specified.